fe/fe-core/src/main/java/com/starrocks/sql/analyzer/ExpressionAnalyzer.java

```diff
@@ -1517,6 +1533,159 @@ public Void visitDefaultValueExpr(DefaultValueExpr node, Scope context) {
+            Column valueColumn = null;
+            for (Column column : dictTable.getBaseSchema()) {
+                if (column.isKey()) {
+                    keyColumns.add(column);
+                }
+                if (column.isAutoIncrement()) {
+                    valueColumn = column;
+                }
+            }
+            // (table, keys..., value_column, strict_mode)
+            int valueColumnIdx;
+            int strictModeIdx;
+            if (params.size() == keyColumns.size() + 1) {
+                valueColumnIdx = -1;
+                strictModeIdx = -1;
+            } else if (params.size() == keyColumns.size() + 2) {
+                if (params.get(params.size() - 1).getType().getPrimitiveType().isStringType()) {
+                    valueColumnIdx = params.size() - 1;
+                    strictModeIdx = -1;
+                } else {
+                    strictModeIdx = params.size() - 1;
+                    valueColumnIdx = -1;
+                }
+            } else if (params.size() == keyColumns.size() + 3) {
+                valueColumnIdx = params.size() - 2;
+                strictModeIdx = params.size() - 1;
+            } else {
+                throw new SemanticException(String.format("dict_mapping function param size should be %d - %d",
+                    keyColumns.size() + 1, keyColumns.size() + 3));
+            }
+
+            String valueField;
+            if (valueColumnIdx >= 0) {
+                Expr valueFieldExpr = params.get(valueColumnIdx);
+                if (!(valueFieldExpr instanceof StringLiteral)) {
+                    throw new SemanticException("dict_mapping function value_column param should be STRING constant");
+                }
+                valueField = ((StringLiteral) valueFieldExpr).getStringValue();
+                valueColumn = dictTable.getBaseColumn(valueField);
+                if (valueColumn == null) {
+                    throw new SemanticException("dict_mapping function can't find value column " + valueField + " in dict table");
+                }
+            } else {
+                if (valueColumn == null) {
+                    throw new SemanticException("dict_mapping function can't find auto increment column in dict table");
+                }
+                valueField = valueColumn.getName();
+            }
+
+            boolean strictMode = false;
+            if (strictModeIdx >= 0) {
+                Expr strictModeExpr = params.get(strictModeIdx);
+                if (!(strictModeExpr instanceof BoolLiteral)) {
+                    throw new SemanticException("dict_mapping function strict_mode param should be bool constant");
+                }
+                strictMode = ((BoolLiteral) strictModeExpr).getValue();
+            }
+            if (!valueColumn.getType().isBigint()) {
+                throw new SemanticException("dict_mapping function value field '" + valueField + "' should be bigint");
+            }
```

Comment:
Why value column must be Bigint type, it's weired and will limit the usage very much.

Reply:
Changed to support all types